commands/command_fsck.go

```diff
@@ -118,20 +140,28 @@ func doFsckObjects(ref *git.Ref) []string {
 	// Attach a filepathfilter to avoid _only_ the excluded paths.
 	gitscanner.Filter = filepathfilter.New(nil, cfg.FetchExcludePaths())
 
-	if err := gitscanner.ScanRef(ref.Sha, nil); err != nil {
-		ExitWithError(err)
+	if start == "" {
+		if err := gitscanner.ScanRef(end, nil); err != nil {
+			ExitWithError(err)
+		}
+	} else {
+		if err := gitscanner.ScanRefRange(start, end, nil); err != nil {
+			ExitWithError(err)
+		}
 	}
 
-	if err := gitscanner.ScanIndex("HEAD", nil); err != nil {
-		ExitWithError(err)
+	if useIndex {
+		if err := gitscanner.ScanIndex("HEAD", nil); err != nil {
+			ExitWithError(err)
+		}
 	}
 
 	gitscanner.Close()
 	return corruptOids
 }
 
 // doFsckPointers checks that the pointers in the given ref are correct and canonical.
-func doFsckPointers(ref *git.Ref) []corruptPointer {
+func doFsckPointers(start, end string, useIndex bool) []corruptPointer {
```

Comment:
Is `useIndex` actually used in this `doFsckPointers()` function, or could it be dropped from the arguments?  (It *is* used in `doFsckObjects()`, obviously.)

Reply:
It can be dropped.